src/Fantomas.Core/Flowering.fs

```diff
@@ -284,13 +284,26 @@ let blockCommentToTriviaInstruction (containerNode: Node) (trivia: TriviaNode) :
 
     match nodeBefore, nodeAfter with
     | Some nb, None when nb.Range.EndLine = trivia.Range.StartLine -> nb.AddAfter(triviaWith false false)
-    | Some nb, Some na when
-        (nb.Range.EndLine < trivia.Range.StartLine
-         && na.Range.StartLine > trivia.Range.EndLine)
-        ->
-        nb.AddBefore(triviaWith true true)
-    | Some nb, _ when nb.Range.EndLine = trivia.Range.StartLine -> nb.AddAfter(triviaWith false false)
     | None, Some na -> na.AddBefore(triviaWith true false)
+    | Some nb, Some na ->
+        if
+            (nb.Range.EndLine < trivia.Range.StartLine
+             && na.Range.StartLine > trivia.Range.EndLine)
+        then
+            // before
+            // (* comment *)
+            // after
+            nb.AddBefore(triviaWith true true)
```

Comment:
I'm lacking the time right now to test this, but shouldn't this be `na.AddBefore()`

Reply:
Good catch, interestingly enough I can remove that entire branch and all tests are still passing.